apps/dashboard/app/lib/smart_attributes/attributes/auto_output_directory.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module SmartAttributes
+  class AttributeFactory
+    # Build this attribute object. Must specify a valid directory in opts
+    #
+    # @param opts [Hash] attribute's options
+    # @return [Attributes::AutoOutputDirectory] the attribute object
+    def self.build_auto_output_directory(opts = {})
+      Attributes::AutoOutputDirectory.new('auto_output_directory', opts)
+    end
+  end
+
+  module Attributes
+    class AutoOutputDirectory < Attribute
+      OUTPUT_LOG_SUFFIX = "/%j-output.log"
```

Comment:
We don't want to provide or dictate suffixes. The whole idea for this feature is that I (as the user) get to specify `%j-output.log` or `%x-output.log` or just `job.out` or whatever I want.

Reply:
ok. I guess my thought process was to let the user choose a directory name in which to contain the log files. Shall I include a key for the slurm defined variables displayed under the select or in a tooltip? Maybe add it to the OOD-Core adapter to establish a pattern for adoption by other schedulers?

Reply:
there are some relatively rigid formatting rules to follow when passing the -o flag

Reply:
A discussion for another ticket.

Reply:
> I guess my thought process was to let the user choose a directory name in which to contain the log files

No, it's to specify the `-o` flag entirely. If they provide something funky, then let the scheduler handle that.

Our business is basically creating GUIs for commands. Once we start dictating what users can and can't do (but could otherwise do in a CLI) we start to get into trouble. 

Inevitably someone's going to want to log to a completely separate directory, and I kinda feel like we should let them.

> Shall I include a key for the slurm defined variables displayed under the select or in a tooltip?

No, this is a medium-advanced use case. If you need/want to reconfigure it, you likely know what you're doing. And if you don't know what you're doing, then Slurm's default should work fine and they never really encounter this at all.

Reply:
I understand. I suppose I am still trying to break my assumption that the user has no technical prowess (as is necessary on the vast majority of projects). I believe that my latest commit satisfies these needs.